circuit/src/threshold/mod.rs

```diff
@@ -1,2 +1,649 @@
+						inst_col_offset + i,
+					)?;
+					sets_pk_y.push(member);
+
+					if i % ADVICE == ADVICE - 1 {
+						ctx.next();
+					}
+				}
+				inst_col_offset += NUM_NEIGHBOURS;
+				ctx.next();
+
+				let mut final_scores = vec![];
+				for i in 0..NUM_NEIGHBOURS {
+					let score = ctx.assign_from_instance(
+						config.common.advice[i % ADVICE],
+						config.common.instance,
+						inst_col_offset + i,
+					)?;
+					final_scores.push(score);
+
+					if i % ADVICE == ADVICE - 1 {
+						ctx.next();
+					}
+				}
+				inst_col_offset += NUM_NEIGHBOURS;
+				ctx.next();
+
+				let target_pk_x = ctx.assign_from_instance(
+					config.common.advice[0], config.common.instance, inst_col_offset,
+				)?;
+				let target_pk_y = ctx.assign_from_instance(
+					config.common.advice[1],
+					config.common.instance,
+					inst_col_offset + 1,
+				)?;
+				inst_col_offset += 2;
+
+				Ok((
+					num_neighbor, init_score, max_limb_value, threshold, score, one, zero,
+					sets_pk_x, sets_pk_y, final_scores, target_pk_x, target_pk_y, inst_col_offset,
+				))
+			},
+		)?;
+
+		// TODO: verify if the "sets" & "final_scores" are valid, using aggregation verify
+
+		// check if the eigentrust score of "target_pk" is the same as "score"
+		let set_pos_chip = SetPositionChip::new(sets_pk_x, target_pk_x);
+		let target_pk_x_idx = set_pos_chip.synthesize(
+			&config.common,
+			&config.set_pos_selector,
+			layouter.namespace(|| "target_pk_x_idx"),
+		)?;
+		let set_pos_chip = SetPositionChip::new(sets_pk_y, target_pk_y);
+		let target_pk_y_idx = set_pos_chip.synthesize(
+			&config.common,
+			&config.set_pos_selector,
+			layouter.namespace(|| "target_pk_y_idx"),
+		)?;
+		layouter.assign_region(
+			|| "target_pk_x_idx == target_pk_y_idx",
+			|region| {
+				let mut ctx = RegionCtx::new(region, 0);
+				let x_idx = ctx.copy_assign(config.common.advice[0], target_pk_x_idx.clone())?;
+				let y_idx = ctx.copy_assign(config.common.advice[1], target_pk_y_idx.clone())?;
+				ctx.constrain_equal(x_idx, y_idx)?;
+				Ok(())
+			},
+		)?;
+
+		let select_item_chip = SelectItemChip::new(final_scores, target_pk_x_idx);
+		let target_pk_score = select_item_chip.synthesize(
+			&config.common,
+			&config.select_item_selector,
+			layouter.namespace(|| "select score from final_scores"),
+		)?;
+		layouter.assign_region(
+			|| "target_pk_score(PI) == score(input)",
+			|region| {
+				let mut ctx = RegionCtx::new(region, 0);
+				let target_pk_score =
+					ctx.copy_assign(config.common.advice[0], target_pk_score.clone())?;
+				let input_score = ctx.copy_assign(config.common.advice[1], score.clone())?;
+				ctx.constrain_equal(target_pk_score, input_score)?;
+				Ok(())
+			},
+		)?;
+
+		// max_score = NUM_NEIGHBOURS * INITIAL_SCORE
+		let max_score = {
+			let mul_chipset = MulChipset::new(num_neighbor, init_score);
+			mul_chipset.synthesize(
+				&config.common,
+				&config.main,
+				layouter.namespace(|| "NUM_NEIGHBOURS * INITIAL_SCORE"),
+			)?
+		};
```

Comment:
No need for new scope because `synthesize` will drop `mul_chipset`

Reply:
Done https://github.com/eigen-trust/protocol/pull/316/commits/6fb493afaea8d9b5e40a7b86460dbb329bc149d0